server/src/controllers/course.ts

```diff
@@ -20,12 +24,54 @@ export interface CourseData {
   evaluationInformation: LocalizedString
 }
 
+export interface InstanceData extends CourseData {
+  startingPeriod: string,
+  endingPeriod: string
+  startDate: Date,
+  endDate: Date,
+  courseType: string,
+  gradingType: string,
+  responsibleTeachers: Array<string>,
+}
+
 export enum Language {
   English = 'EN',
   Finnish = 'FI',
   Swedish = 'SV'
 }
 
+function parseSisuInstance (instance: SisuInstance): InstanceData {
+  return {
+    // instance id here is Sisu instance id (string)
+    id: instance.id,
+    courseCode: instance.code,
+    minCredits: instance.credits.min,
+    maxCredits: instance.credits.max,
+    department: {
+      en: instance.organizationName.en,
+      fi: instance.organizationName.fi,
+      sv: instance.organizationName.sv
+    },
+    name: {
+      en: instance.name.en,
+      fi: instance.name.fi,
+      sv: instance.name.sv
+    },
+    evaluationInformation: {
+      en: instance.summary.assesmentMethods.en,
+      fi: instance.summary.assesmentMethods.fi,
+      sv: instance.summary.assesmentMethods.sv
+    },
+    startingPeriod: '-',
+    endingPeriod: '-',
+    startDate: instance.startDate,
+    endDate: instance.endDate,
+    courseType: instance.type === 'exam-exam' ? 'EXAM' : 'LECTURE',
+    gradingType: instance.summary.gradingScale.fi === '0-5' ? 'NUMERICAL' : 'PASSFAIL',
```

Comment:
This may be fine for now but we should (maybe in the next sprint) use the enums from #116. I would add a TODO reminder to fix this.

Reply:
added reminder https://github.com/aalto-grades/base-repository/pull/126/commits/d9bb67b83c330f34b6a651500784ddca26586d13